invenio_rdm_records/requests/access/models.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Database models related to access requests."""
+
+import uuid
+from datetime import date, datetime
+from secrets import token_urlsafe
+
+from flask import current_app
+from invenio_db import db
+from sqlalchemy_utils import UUIDType
+
+from .permissions import AccessRequestTokenNeed
+
+
+class AccessRequestToken(db.Model):
+    """Token for accessing guest-based access requests.
+
+    These tokens are used in the guest-based access request workflow to
+    signify that somebody has requested access to a record via e-mail,
+    but hasn't verified their e-mail yet.
+    Once the e-mail address is verified to create the access request, this
+    the corresponding ``AccessRequestToken`` will be deleted, and a new
+    guest access request will be created with some of the token's information.
+    """
+
+    __tablename__ = "rdm_records_access_request_tokens"
+
+    id = db.Column(UUIDType, primary_key=True, default=uuid.uuid4)
+    """Access request token ID."""
+
+    token = db.Column(db.String, nullable=False)
+    """Randomly generated token which will grant access to the request."""
+
+    created = db.Column(
+        db.DateTime, nullable=False, default=datetime.utcnow, index=True
+    )
+    """Creation timestamp."""
+
+    expires_at = db.Column(db.DateTime, nullable=False)
+    """Expiration date."""
+
+    email = db.Column(db.String, nullable=False)
+    """E-mail address for which the access request was made."""
+
+    full_name = db.Column(db.String, nullable=False, default="")
```

Comment:
It should be fine – we're not allowing null values to be stored in the DB, and the default value for that field is set to the empty string rather than null.